### ISSUE TITLE
Libusb1 Async Control Buffer Null

### ISSUE THREAD

**Issue by jdkeenan** [vpelletier/python-libusb1#27]
I've been using libusb1 for around 9 months for multiple usb devices and found out ew 1.6.0 versions and later cause problems with my devices. I'm pretty sure it's tied to this issue (https://github.com/vpelletier/python-libusb1/issues/21) and using the getBuffer() command to get the most recent data sent when preforming a setControl call.
After some further inspection I couldn't find where the buffer is actually updated during a 0xC1 request. Any help would be much appreciated.

Here is some example code. I'm awaiting asyncControlTransfer which is calling setControl. This works perfectly fine with 1.5.3
```
transfer = await asyncusb.asyncControlTransfer(
            self.handle, 0xC1,
            0x20,
            0, 0, 8, timeout = self.CT_TIMEOUT)
result = transfer.getUserData()

def asyncControlTransfer(handle, *args, **kwargs):
    ct = handle.getTransfer()
    ct.setControl(*args, **kwargs)
    return asyncTransfer(ct)

def asyncTransfer(transfer):
    fut = asyncio.Future()
    def callback(t):
        try:
            result = t.getStatus()
            if result != TRANSFER_COMPLETED:
                raise USBTransferError(result)
            if not fut.cancelled():
                fut.set_result(t)
        except Exception as e:
            fut.set_exception(e)
    transfer.setCallback(callback)
    result = transfer.submit()
    return fut
```

**Comment by vpelletier** [comment]
The idea behind the change in #21 is that `__transfer_py_buffer` gets modified when the memory behind `__transfer_buffer` is modified inside the C library, to avoid copies.

But I broke this with this line in setControl, which is likely the cause of this bug:

```python
self.__transfer_py_buffer = transfer_py_buffer[CONTROL_SETUP_SIZE:]
```

Could you change this line to the following and report if it fixes the problem ?
```python
self.__transfer_py_buffer = memoryview(transfer_py_buffer)[CONTROL_SETUP_SIZE:]
```

The problem with this fix is that, while [python3's bytearray only accepts integers for item assignment, python2's memoryview rejects integers](http://bugs.python.org/issue29404) (I'm surprised to be apparently the first to report this on python's bugtracker given the number of google hits this issue has...). So if I apply it, I force the user of my module to be actively incompatible... Gah.

**Comment by jdkeenan** [comment]
Awesome, this solved the problem.
Yeah, this is a pretty amazing package and has been incredible useful in my work, so keep up the great work! Also thanks for the quick response!

**Comment by vpelletier** [comment]
Thanks for the warm words, and happy my work is useful to you !

I reopen this issue, as the bug is still present in master - just so I do not forget to push the fix once I know what to expect from python 2.7.

**Comment by vpelletier** [comment]
Now fixed in master, and released on pypi as 1.6.4, closing.
